### PR TITLE
Integration tests for secrets rotation

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

Implement end-to-end tests for secrets rotation in secrets provider.

### Implemented Changes

- Added test case for secrets rotation

### Connected Issue/Story

CyberArk internal issue link: [ONYX-16912](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-16912)

### Definition of Done

- [x] Tests exist for secrets rotation

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [x] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: deploy/test/test_cases/TEST_ID_28_secrets_rotation.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
```
Comment (rpothier) [inline deploy/test/test_cases/TEST_ID_28_secrets_rotation.sh:1]: This has a lot of duplication with test 27, could we make some common functions for each test to call?

**Review by rpothier — COMMENTED** [review]
Looks good, just one comment.

File: deploy/utils.sh [hunk 2]
```diff
@@ -571,3 +571,32 @@ wait_for_job() {
...
+    expected_num_replicas=`$CONFIG_DIR/$test_yaml_file_name.sh.yml |  awk '/replicas:/ {print $2}' `
+    
+    # Deployment (Deployment for k8s and DeploymentConfig for Openshift) might fail on error flows, even before creating the pods. If so, re-deploy.
+    if [[ "$PLATFORM" = "kubernetes" ]]; then
+        $cli_with_timeout "get deployment $deployment_name -o jsonpath={.status.replicas} | grep '^${expected_num_replicas}$'|| $cli_without_timeout rollout latest deployment $deployment_name"
+    elif [[ "$PLATFORM" = "openshift" ]]; then
+        $cli_with_timeout "get dc/$deployment_name -o jsonpath={.status.replicas} | grep '^${expected_num_replicas}$'|| $cli_without_timeout rollout latest dc/$deployment_name"
+    fi
+
+    echo "Expecting for $expected_num_replicas deployed pods"
```
Comment (rpothier) [inline deploy/utils.sh:599]: Should this just be Expecting $expected ... ?

File: deploy/utils.sh [hunk 3]
```diff
@@ -571,3 +571,32 @@ wait_for_job() {
...
+    $cli_with_timeout "get pods --namespace=$APP_NAMESPACE_NAME --selector app=$deployment_name --no-headers | wc -l | grep $expected_num_replicas"
+  fi
+}
```
Comment (rpothier) [inline deploy/utils.sh:602]: Can you add a new line?

**Review by rpothier — COMMENTED** [review]
Looks great! just a couple minor nits.

File: deploy/utils.sh (hunk 2 above)
Comment (szh) [reply deploy/utils.sh:599]: Probably. I really copied this from the 27 script but I'll fix it.

**Review by rpothier — APPROVED** [review]
LGTM!

**Pull request merged**